docs/docsite/rst/dev_guide/developing_modules_documenting.rst

```diff
@@ -10,9 +12,100 @@ which conforms to the schema defined below. You may find it easier to
 start writing your ``DOCUMENTATION`` string in an editor with YAML
 syntax highlighting before you include it in your Python file.
 
+All modules must have the following sections defined in this order:
+
+1. ANSIBLE_METADATA
+2. DOCUMENTATION
+3. EXAMPLES
+4. RETURNS
+5. Python imports
+
+.. note:: Why don't the imports go first?
+
+  Keen Python programmers may notice that contrary to PEP8's advice we don't put ``imports`` at the top of the file. This is because the ``ANSIBLE_METADATA`` through ``RETURNS`` sections are not used by the module code itself; they are essentially extra docstrings for the file. The imports are placed after these special variables for the same reason as PEP8 puts the imports after the introductory comments and docstrings. This keeps the active parts of the code together and the pieces which are purely informational apart. The decision to exclude E402 is based on readability (which is what PEP8 is about). Documentation strings in a module are much more similar to module level docstrings, than code, and are never utilized by the module itself. Placing the imports below this documentation and closer to the code, consolidates and groups all related code in a congruent manner to improve readability, debugging and understanding.
+
+.. warning:: Why do some modules have imports at the bottom of the file?
+
+  If you look at some existing older modules, you may find imports at the bottom of the file. Do not copy that idiom into new modules as it is a historical oddity due to how modules used to be combined with libraries. Over time we're moving the imports to be in their proper place.
+
+
+
+ANSIBLE_METADATA Block
+----------------------
+
+ANSIBLE_METADATA contains information about the module for use by other tools. At the moment, it informs other tools which type of maintainer the module has and to what degree users can rely on a module's behaviour remaining the same over time.
+
+For new modules, the following block can be simply added into your module::
+
+   ANSIBLE_METADATA = {'status': ['preview'],
+                       'supported_by': 'community',
+                       'version': '1.0'}
+
+.. warning:: version field
+
+   This is the version of the ``ANSIBLE_METADATA`` schema, *not* the version of the module.
+
+.. warning::
+
+   Promoting a module's ``status`` or ``supported_by`` status should only be done by members of the Ansible Core Team.
+
+Version 1.0 of the metadata
++++++++++++++++++++++++++++
+
+Structure
+`````````
+
+Format::
+
+  ANSIBLE_METADATA = {
+      'version': '1.0',
+      'supported_by': 'core|community|core_curated',
```

Comment:
Not that it is necessarily the source of truth, but validate-modules sets the schema to this field as:

```
        Required('supported_by'): Any('core', 'community', 'unmaintained',
                                      'committer')
```

This was done based off of the proposal, but I also see that the metadata tool lists:

core, curated, committer, community